server/index.js

```diff
@@ -64,6 +64,107 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const sql = `
+  select "c"."cartItemId",
+       "c"."price",
+       "p"."productId",
+       "p"."image",
+       "p"."name",
+       "p"."shortDescription"
+  from "cartItems" as "c"
+  join "products" as "p" using ("productId")
+ where "c"."cartId" = $1
+  `;
+  const value = [`${req.session.cartId}`];
+  if (req.session.cartId) {
+    db.query(sql, value)
+      .then(response => {
+        res.status(200).json(response.rows);
+      })
+      .catch(err => next(err));
+  } else {
+    return [];
+  }
+
+});
+
+app.post('/api/cart/:productId', (req, res, next) => {
+  const { productId } = req.params;
+
+  if (productId <= 0) {
+    return res.status(400).json({ error: 'productId must be a positive integer' });
+  }
+
+  const sqlInsert = `
+  insert into "carts" ("cartId", "createdAt")
+  values (default, default)
+  returning "cartId"
+  `;
+  const sql = `
+  select "products"."price"
+    from "products"
+   where "productId" = $1
+  `;
+  const values = [`${productId}`];
+  // first chain
+  db.query(sql, values)
+    .then(response => {
+
+      if (!response.rows[0]) {
+        return res.status(400).json({ error: `productId ${productId} does not match any existing products` });
```

Comment:
You need to `throw new ClientError` here (and pass the message and status code) to jump down to the `.catch` so the other `.then()`s don't try to run.